doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Scorecard Custom Test Framework
+authors:
+  - "@jemccorm"
+reviewers:
+  - TBD
+  - "@joelandford"
+  - "@zeus"
+approvers:
+  - TBD
+  - "@shurley"
+  - "@dmesser"
+creation-date: 2020-03-19
+last-updated: 2020-03-19
+status: implementable
+---
+
+# Scorecard - Custom Tests
+
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created 
+
+## Summary
+
+This enhancement proposal outlines the SDK scorecard’s design for running of custom tests, specifically tests that are outside of what Red Hat includes by default within the scorecard’s tests.   The proposed design would provide both a means to write custom tests and also convert the existing scorecard tests (basic and olm) into the custom test format.
+
+## Motivation
+
+The scorecard needs a robust means of enabling end-users of the SDK to write and execute their own operator tests.   Ideally, the custom test framework would decouple test logic from within the SDK itself, making the SDK and scorecard logic in particular more flexible and maintainable in the long run.
+
+## Goals
+
+ * Document a design for scorecard to execute end-user or custom tests.
+
+ * Document stories/steps required to deliver scorecard custom test functionality.
+
+### Non-Goals
+
+ * This is a design proposal and does not include a proof-of-concept.
+
+## Proposal
+
+This design proposal is to establish a pattern for implementing custom scorecard tests and also support the goal of migrating existing scorecard tests into the same format as the custom tests.
+
+With this proposal, the scorecard will continue to setup and teardown the Operator being tested, as well as creating Custom Resources being tested.
+
+### User Stories
+
+#### Story 1
+
+Scorecard users will be able to construct custom tests and execute them using the scorecard.
+
+#### Story 2
+
+Scorecard will ship with an example custom test that end users could use as a reference when writing their custom tests.
+
+#### Story 3
+
+Scorecard internal tests will migrate to the custom test format and therefore be externalized from the scorecard binary itself.
```

Comment:
Why it is important for the dev? What is the goal with it? I mean, what will be possible to achieve with?

Reply:
I'll make a note about this, thx.